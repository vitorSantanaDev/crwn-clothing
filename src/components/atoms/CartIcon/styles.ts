import styled from 'styled-components'

export const CartIconWrapper = styled.div`
  width: 45px;
  height: 45px;
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  cursor: pointer;

  .shoppingIcon {
    width: 24px;
    height: 24px;
  }
`
export const ItemsNumber = styled.span`
  position: absolute;
  font-size: 12px;
  font-weight: bold;
  bottom: 12px;
`
