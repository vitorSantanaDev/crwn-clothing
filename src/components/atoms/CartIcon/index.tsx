import { useContext } from 'react'

import { CartContext } from 'contexts'

import { ReactComponent as ShoppingIcon } from '../../../assets/shopping-bag.svg'

import * as S from './styles'

export default function CartIcon() {
  const { isCartOpen, setIsCartOpen, cartCount } = useContext(CartContext)

  const toggleIsCartOpen = () => setIsCartOpen?.(!isCartOpen)

  return (
    <S.CartIconWrapper onClick={toggleIsCartOpen}>
      <ShoppingIcon className="shoppingIcon" />
      <S.ItemsNumber>{cartCount}</S.ItemsNumber>
    </S.CartIconWrapper>
  )
}
