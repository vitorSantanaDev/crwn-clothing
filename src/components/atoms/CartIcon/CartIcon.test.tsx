import { render } from '@testing-library/react'
import CartIcon from '.'

describe('<ButtonComponent />', () => {
  it('should render the cart icon', () => {
    render(<CartIcon />)

    const icon = document.querySelector('.shoppingIcon')
    expect(icon).toBeInTheDocument()
  })
  it('should match snapshot', () => {
    const { container } = render(<CartIcon />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
