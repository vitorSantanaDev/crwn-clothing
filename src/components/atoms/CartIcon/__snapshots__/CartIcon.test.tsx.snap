// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonComponent /> should match snapshot 1`] = `
<div
  class="sc-bczRLJ koYlEn"
>
  <svg
    class="shoppingIcon"
  >
    shopping-bag.svg
  </svg>
  <span
    class="sc-gsnTZi ccslnz"
  >
    0
  </span>
</div>
`;
